src/common/snippets/include/snippets/op/broadcastmove.hpp

```diff
@@ -38,7 +39,7 @@ class BroadcastMove : public ov::op::Op {
     };
 
 protected:
-    ov::PartialShape output_shape;
+    ov::Dimension bcast_dimension;
```

Comment:
I discussed this offline with @IvanNovoselov: since BroadcastLoad/BroadcastMove nodes can broadcast only the last dimension, there is no need to store the entire target `PartialShape`. So we decided to store only target dimension

Reply:
Hmm, I have just a question...
At the moment we support even `Broadcast` op in MHA pattern. But with special [condition](https://github.com/openvinotoolkit/openvino/blob/523333cf1d1688a14d4fa4e5952e58de76f8ab5b/src/common/snippets/src/pass/mha_tokenization.cpp#L51). If we'd like to support any `Broadcast` op (we have these plans as I remember... in the nearest (haha) future), we should have opportunity to infer shapes. I'm afraid of that we may loss this opportunity now. What do you think?

@IvanNovoselov

Reply:
I think that the actual problem is in [BroadcastToMoveBroadcast](https://github.com/v-Golubev/openvino/blob/9fafcabb7c7b221821b0b77616d9967c8d13223f/src/common/snippets/include/snippets/pass/broadcast_to_movebroadcast.hpp#L14-L19) pass:
1) `MoveBroadcast` op must be inserted ONLY if the last dim is broadcasted, since we need a special instruction in this case (that's satisfied)
2) Otherwise we need to retain `Broadcast` operation to guarantee correct shape flow (that's violated). This op should be mapped to `Nop` emitter to produce no code.